### PR TITLE
Enhance OpenShift Pipelines installation depending on environment + pipeline route bugfix

### PR DESCRIPTION
**Pull request by jgarciao**
- When installing Red Hat OpenShift Pipelines operator, select version depending on OpenShift version, platform (AWS/GCP/...) and RHODS flavor (Managed/Self-Managed)
- Do not deploy KubeflowUI when creating a pipeline server with the sample DataSciencePipelineApplication
- Fix route used when creating a pipeline server with the sample DataSciencePipelineApplication (same as in PR #809)

### CODE REVIEW

**Review by diegolovison — APPROVED** [review]

File: ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource [hunk 1]
```diff
@@ -10,17 +11,44 @@ Library             ../../libs/DataSciencePipelinesAPI.py
...
+    ${platform}=    Fetch Cluster Platform Type
+    IF    ${oc_version} == "4.10"
+        ${openshift_pipelines_version}=    Set Variable    pipelines-1.8"
+    ELSE IF    ${oc_version} == "4.11"
+        ${openshift_pipelines_version}=    Set Variable    pipelines-1.9
+    ELSE
+        IF    ${is_self_managed} == ${TRUE} and "${platform}" == "AWS"
+            ${openshift_pipelines_version}=    Set Variable    pipelines-1.10
+        ELSE IF    ${is_self_managed} == ${FALSE} and "${platform}" == "AWS"
+            ${openshift_pipelines_version}=    Set Variable    pipelines-1.11
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource:37]: why SM and CS requires 2 different versions of the pipelines operator?

Comment (bdattoma) [inline ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource:37]: why SM and CS requires 2 different versions of the pipelines operator?

Comment (jgarciao) [reply ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource:37]: They don't. This logic is to ensure that we test OpenShift Pipelines 1.10 and 4.11 in AWS and GCP without having to run all possible combinations

**Comment by jgarciao** [comment]
Holding the merge after 1.28.1 release

File: ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource (hunk 1 above)
Comment (jgarciao) [reply ods_ci/tests/Resources/Page/Operators/OpenShiftPipelines.resource:40]: I've updated the [Documentation] section to explain this

**Review by bdattoma — APPROVED** [review]

**Review by diegolovison — APPROVED** [review]

**Pull request merged**